whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/RpslCharset.java

```diff
@@ -0,0 +1,36 @@
+package net.ripe.db.whois.common.rpsl;
+
+import com.google.common.base.Splitter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.charset.UnsupportedCharsetException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public enum RpslCharset {
```

Comment:
Do we need a wrapper around `Charset` ?

Reply:
yes, because
Make sure "UTF8", "UTF-8", "LATIN1" and "LATIN-1" are supported (We will allow all of these, even if they are not standard in Java)

And I think that an ENUM solves this use case easily

Reply:
Assuming that we can't add charset aliases at runtime (I don't know?), then I suggest simplifying the wrapper class:

(1) call it something like "WhoisCharsets" or "CharsetWrapper" instead of "Rpsl" as it's not related to RPSL
(2) Default to Charsets.forName
(3) No need for an enum, as we're only mapping a small number of exceptions (maybe only 1):

```
public static Charset forName(final String charsetName) {
    switch (charsetName.toUppercase()) {
        case "LATIN-1": return StandardCharsets.ISO_8859_1;
        default: return Charsets.forName(charsetName);
    }
}
```

Reply:
Talked via Mattermost